Rakefile

```diff
@@ -262,6 +263,399 @@ task :xcode => [:dependencies] do
+
+  #confirm that GPG tools is installed
+  task :check do
+    puts "Checking system for GPG Tools"
+    unless command?("gpg")
+      Rake::Task["gpg_key:install"].invoke
+    else
+      puts "GPG Tools found"
+    end
+  end
+
+  #install GPG Tools
+  task :install do
+    puts "GPG Tools not found.  Installing GPG Tools"
+    sh "brew install gpg"
+  end
+
+  #Ask developer if they need to create a new key.
+  #If yes, begin process of creating key, if no move on
+  task :prompt do
+    if create_gpg_key?
+      if create_default_key?
+        display_default_config_helpers
+        Rake::Task["gpg_key:generate_default"].invoke
+      else
+        Rake::Task["gpg_key:generate_custom"].invoke
+      end
+    else
+      next
+    end
+
+  end
+
+  #Generate new GPG key
+  task :generate_custom do
+    puts ""
+    puts "Begin Generating Custom GPG Keys"
+    puts "====================================================================================="
+
+    sh "gpg --full-generate-key", verbose: false
+  end
+
+  #Generate new default GPG key
+  task :generate_default do
+    puts ""
+    puts "Begin Generating Default GPG Keys"
+    puts "====================================================================================="
+
+    sh "gpg --generate-key", verbose: false
+  end
+
+  #prompt developer to send GPG key to Platform
+  task :finish do
+    puts "====================================================================================="
+    puts "Key Generation Complete!"
+    puts "Please send your GPG public key to Platform 9-3/4"
+    puts "You can contact them in the Slack channel #platform9"
+    puts "====================================================================================="
+  end
+
+  #ask user if they want to create a key, loop till given a valid answer
+  def create_gpg_key?
+    puts "====================================================================================="
+    puts "To access production credentials for the WordPress app you will need to a GPG Key"
+    puts "Do you need to generate a new GPG Key?"
+    puts "Press 'Y' to create a new key.  Press 'N' to skip"
+
+    display_prompt_response
+  end
+
+  #ask user if they want to create a key,  loop till given a valid answer
+  def create_default_key?
+    puts "====================================================================================="
+    puts "You can choose to setup with a default or custom key pair setup"
+    puts "Default setup - Type: RSA to RSA, RSA length: 2048, Valid for: does not expire"
+    puts "Would you like to continue with the default setup?"
+    puts "====================================================================================="
+    puts "Press 'Y' for Yes.  Press 'N' for custom configuration"
+
+    display_prompt_response
+  end
+
+  #prompt for a Y or N response, continue asking if other character
+  #return true for Y and false for N
+  def display_prompt_response
+    response = STDIN.gets.strip.upcase
```

Comment:
Nice touch using the `.upcase` on the input 👍

Reply:
Thanks!